docs/development/concepts/hotwire-view-components/README.md

```diff
@@ -0,0 +1,275 @@
+
+    journal.update(journal_params) # in real life this is done through a service obviosuly ;)
+
+    # update the journal show component
+    update_via_turbo_stream(
+      component: JournalShowComponent.new(journal: journal)
+    )
+
+    # respond with turbo streams which were collected in the @turbo_streams variable behind the scenes
+    # handy if this method is just meant to respond to turbo-stream requests
+    respond_with_turbo_streams
+  end
+
+  def create
+    journal = Journal.create(journal_params) # in real life this is done through a service obviosuly ;)
+
+    if journal.errors.empty?
+      # append the new model to the index component
+      # prepend is also possible
+      append_via_turbo_stream(
+        component: JournalShowComponent.new(journal: journal),
+        target_component: JournalIndexComponent.new(work_package: @work_package)
+      )
+      # Note: the target_component does not get rendered
+      # the instatiation is just required for the turbo-stream generation
+
+      # you can use multiple turbo_stream methods in one controller action
+      # e.g. update the new component to render an initial form
+      update_via_turbo_stream(
+        component: JournalNewComponent.new(work_package: journal.work_package)
+      )
+    else
+      # optionally set a turbo status for the response
+      @turbo_status = :bad_request
+
+      # trigger a flash message via turbo-stream
+      # more on this here lookbook/pages/patterns/flash_banner
+      update_flash_message_via_turbo_stream(
+        message: journal.errors.full_messages.join(", "),
+        scheme: :danger
+      )
+    end
+
+    # respond with turbo streams which were collected in the @turbo_streams variable behind the scenes
+    # handy if this method is just meant to respond to turbo-stream requests
+    respond_with_turbo_streams
+  end
+
+  # ...
+end
+```
+
+## Mixing turbo-streams and other responses
+
+TODO: Discuss the below example
+
+```ruby
+class JournalController < ApplicationController
+  include OpTurbo::ComponentStream # include this module!
+
+  # ...
+
+  def update
+    # ...
+
+    respond_to do |format|
+      format.html do
+        # ...
+      end
+      format.turbo_stream do
+        update_via_turbo_stream(
+          component: JournalShowComponent.new(journal: journal)
+        )
+
+        render turbo_stream: turbo_streams, status: :ok
+      end
+    end
+  end
+
+  # ...
+end
+```
+
+## Usage alongside Primer Forms/Buttons
+
+TODO: is `turbo: true` required here?
+
+```ruby
+<%=
+  component_wrapper do
+    # ...  
+    primer_form_with(
+      model: journal,
+      method: :put,
+      data: { turbo: true, turbo_stream: true }, # add this!
+      url: journal_path(id: journal.id)
+    ) do |f|
+      # ...
+    end
+    # ...
+  end
+%>
+```
+
+```ruby
+<%=
+  component_wrapper do
+    # ...  
+    render(Primer::Beta::Button.new(
+      scheme: :secondary,
+      size: :medium,
+      tag: :a,
```

Comment:
```suggestion
```

These parameters are a distraction as they are not needed for understanding how Hotwire and Primer forms work together. Better remove them.